src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -277,14 +276,20 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             }
 
             var oldContainingSymbol = _factory.CurrentFunction;
+            var oldIsCodeCoverageDisabled = _factory.IsCodeCoverageDisabled;
             try
             {
                 _factory.CurrentFunction = localFunction;
+                if (!oldIsCodeCoverageDisabled)
+                {
+                    _factory.IsCodeCoverageDisabled = localFunction.IsDirectlyExcludedFromCodeCoverage;
```

Comment:
>IsCodeCoverageDisabled [](start = 29, length = 22)

After thinking about the problem more, this approach feels somewhat fragile. It relies on the fact that factory is shared between DynamicAnalysisInjector and the LocalRewriter. I believe that up until now, that was just an optimization, not a correctness requirement. Also, SyntheticBoundNodeFactory is often getting cloned, for the purpose of using different syntax node, for example, etc. The more context is kept in the factory, the harder it is for one to decide what part of context is supposed to be preserved in each particular situation and what is the right way to preserve it. I would like to suggest an alternative approach. What if we simply adjusted/restored the ```_instrumenter``` around visiting the body of the local function?